server/server.js

```diff
@@ -26,6 +30,194 @@ app.get('/api/hello', (req, res) => {
+    res.status(201).json(user);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.post('/api/users/log-in', async (req, res, next) => {
+  try {
+    const { username, password } = req.body;
+    if (!username || !password) { throw new ClientError(401, 'Invalid Login'); }
+    const sql = `
+    select "userId",
+     "hashedPassword"
+      from "users"
+      where "username" = $1
+    `;
+    const params = [username];
+    const result = await db.query(sql, params);
+    if (!result) { throw new ClientError(401, 'Invalid Login'); }
+    const user = result.rows[0];
+    console.log('log-in');
+    if (!user) { throw new ClientError(401, 'Invalid Login'); }
+    const { userId, hashedPassword } = user;
+    if (!await argon2.verify(hashedPassword, password)) { throw new ClientError(401, 'Invalid Login'); }
+    const payload = { userId, user };
+    const token = jwt.sign(payload, process.env.TOKEN_SECRET);
+    res.json({ user: payload, token });
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.use(authorizationMiddleware);
+
+app.post('/api/images/upload', async (req, res, next) => {
+  try {
+    const { caption } = req.body;
+    const { userId } = req.user;
+    if (!caption) { throw new ClientError(400, 'caption is a required field'); }
+    const url = `/images/${req.file.filename}`;
+    const sql = `
+    insert into "images" ("userId", "url", "caption")
+    values ($1, $2 ,$3)
+    returning *
+  `;
+    const params = [userId, url, caption];
+    const result = await db.query(sql, params);
+    const image = result.rows[0];
+    res.status(201).json(image);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.get('/api/images/', async (req, res, next) => {
+  try {
+    const { userId } = req.user;
+    const sql = `
+    select "caption"
+    from "images"
+    where "userId" = $1
+    `;
+    const params = [userId];
+    const result = await db.query(sql, params);
+    const images = result.rows[0];
+    res.status(200).json(images);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.get('/api/images/:imageId', async (req, res, next) => {
+  try {
+    const { imageId } = req.user;
+    const sql = `
+    select *
+    from "images"
+    where "imageId" = $1
+    `;
+    const params = [imageId];
+    const result = await db.query(sql, params);
+    const image = result.rows[0];
+    res.status(200).json(image);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.delete('/api/images/:imageId', async (req, res, next) => {
+  try {
+    const { imageId } = req.user;
+    const sql = `
+    delete
+      from "images"
+      where "imageId" = $1
+      returning *
+      `;
+    const params = [imageId];
+    const result = await db.query(sql, params);
+    const image = result.rows[0];
+    image ? res.status(204).json(image) : res.status(404).json({ error: `Cannot find image with imageId ${imageId}` });
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.post('/api/songs/upload', async (req, res, next) => {
```

Comment:
You'll need uploads middleware here, too. And we'll have to do some work with it so that these uploads go into the `/songs` directory. See me when you get there if you need help.